### PR TITLE
Change tx driver handling.

### LINKED ISSUE #10191: Quarkus-based distribution cannot use MariaDB with Galera replication due to XA transactions being unsupported
### Describe the bug

Using the Quarkus-based version of Keycloak 17.0.0 with a MariaDB backend with Galera replication enabled fails to start. Galera does not yet support XA transactions which Keycloak uses.

### Version

17.0.0

### Expected behavior

Keycloak should start normally and create a database.

### Actual behavior

Keycloak fails to start with
```
WARN  [com.arjuna.ats.jta] (main) ARJUNA016061: TransactionImple.enlistResource - XAResource.start returned: ARJUNA016099: Unknown error code:0 for < formatId=131077, gtrid_length=35, bqual_length=36, tx_uid=0:ffffac120005:92f9:620a53be:0, node_name=quarkus, branch_uid=0:ffffac120005:92f9:620a53be:3, subordinatenodename=null, eis_name=0 >: javax.transaction.xa.XAException: (conn=140) This version of MariaDB doesn't yet support 'XA transactions with Galera replication'
```

### How to Reproduce?

- Set up a MariaDB with Galera replication enabled. No actual clustering is needed.
- Build Keycloak with `--db=mariadb`
- Start Keycloak using the Galera-replicated MariaDB as database

### Anything else?

I've tried disabling XA transactions using `quarkus.datasource.jdbc.transactions=enabled` in `conf/quarkus.properties` before building Keycloak, but still get the same error.

### PR DESCRIPTION
**Pull request by DGuhr**
Introduce a non-tx driver for the vendors and map based on new build option transaction-tx-enabled

Closes #10191 and others.

### CODE REVIEW

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ConfigCategory.java [hunk 1]
```diff
@@ -4,13 +4,14 @@
...
     CLUSTERING("Cluster", 10),
     DATABASE("Database", 20),
-    FEATURE("Feature", 30),
-    HOSTNAME("Hostname", 40),
-    HTTP("HTTP/TLS", 50),
-    METRICS("Metrics", 60),
-    PROXY("Proxy", 70),
-    VAULT("Vault", 80),
-    LOGGING("Logging", 90),
+    TRANSACTION("Transactions",30),
```
Comment (pedroigor) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ConfigCategory.java:7]: We don't have any other category in the plural, right? Isn't better `transaction`?

**Review by pedroigor — COMMENTED** [review]
What about the database tests? We have one using non-xa for MSSQL and I think it should be updated.

And we probably want to update another database test too such as mysql or postgres?

**Comment by DGuhr** [comment]
@pedroigor for updating existing test: sure. but the scheme is the same, we have a configuration test covering the mechanism and one "smoke" test using mysql with a non-xa driver. imo this suffices, but yes I can add more tests. Then again, a meaningful test would be to really spin up a database like mariadb_galera in a testcontainer that can only be used with non-xa. Should I work on that instead? Shouldn't be that hard I think. I have a working setup here: https://gist.github.com/DGuhr/59c17fc6ec461a5ec3421fa4bb8eacd8 I just want to avoid running too many tests testing kind of the same thing, as the time of the testsuite run is pretty high. but not sure.

File: docs/guides/src/main/server/db.adoc [hunk 2]
```diff
@@ -5,7 +5,7 @@
 <@tmpl.guide
     title="Configuring the database"
     summary="An overview about how to configure relational databases"
-    includedOptions="db db-*">
+    includedOptions="db db-* transactions-xa-enabled">
```
Comment (pedroigor) [inline docs/guides/src/main/server/db.adoc:8]: You changed the category name to be in the singular but the option is still in the plurar. To keep consistency, please use the singular version.

**Comment by pedroigor** [comment]
@DGuhr More tests better. But we can do that as a follow-up because the mechanics are already being covered at some degree.

File: docs/guides/src/main/server/db.adoc (hunk 2 above)
Comment (DGuhr) [reply docs/guides/src/main/server/db.adoc]: Thanks for noticing, changed in all places :)

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java [hunk 3]
```diff
@@ -106,6 +103,18 @@ private DatabasePropertyMappers(){}
         };
     }
 
+    private static String getXaOrNonXaDriver(String db, ConfigSourceInterceptorContext context) {
+        ConfigValue xaEnabledConfigValue = context.proceed("kc.transaction-xa-enabled");
+
+        boolean isXaEnabled = xaEnabledConfigValue == null || Boolean.parseBoolean(xaEnabledConfigValue.getValue());
+
+        if (isXaEnabled) {
```
Comment (pedroigor) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java:111]:
```suggestion
        return Database.getDriver(db, isXaEnabled).orElse(db);
```

Should be simpler. But not a blocker.

**Review by pedroigor — DISMISSED** [review]

**Review by pedroigor — APPROVED** [review]

**Comment by mcattle** [comment]
Is there a (rough) ETA on the next release that includes this change?

**Comment by DGuhr** [comment]
@mcattle rough ETA is always hard. but considering we did most of the [backports](https://github.com/keycloak/keycloak/commits/17.0) for 17.0.1 yesterday, including this fix, let's say "shortly. very shortly." :)

**Pull request merged**